pandas/plotting/_matplotlib/core.py

```diff
@@ -969,6 +966,16 @@ def _make_plot(self):
         else:
             c_values = c
 
+        # plot a colorbar only if a colormap is provided or necessary
```

Comment:
maybe remove this line and add a more descriptive comment for the next line (`c_is_column_not_containing_colors`)? this comment does not seem to add too much value to help others understand?